### PR TITLE
Fix tab index for pdf options

### PR DESCRIPTION
**Pull request by JaneHazen**

### CODE REVIEW

**Comment by mttwc** [comment]
Nice! Let's check with PM to see if the new navigation order (i.e., navigating to and from the Less button) is suitable

File: src/scripts/clipperUI/components/previewViewer/previewViewerHeaderComponentBase.tsx [hunk 1]
```diff
@@ -25,7 +25,7 @@ export interface ControlGroup {
  * of buttons or similar header entities. Child classes need to simply declare each control's
  * buttons.
  */
-export abstract class PreviewViewerHeaderComponentBase<T, P> extends ComponentBase<T, P> {
+export abstract class ButtonGroupingsComponentBase<T, P> extends ComponentBase<T, P> {
```
Comment (mttwc) [inline src/scripts/clipperUI/components/previewViewer/previewViewerHeaderComponentBase.tsx]: Why was this renamed?

File: src/scripts/clipperUI/components/previewViewer/previewViewerHeaderComponentBase.tsx [hunk 2]
```diff
@@ -38,11 +38,22 @@ export abstract class PreviewViewerHeaderComponentBase<T, P> extends ComponentBa
 		let buttonGroups = this.getControlGroups();
 
 		for (let i = 0; i < buttonGroups.length; i++) {
-			let id = buttonGroups[i].id;
-			let className = buttonGroups[i].className;
+			let currentButtonGroup = buttonGroups[i];
```
Comment (mttwc) [inline src/scripts/clipperUI/components/previewViewer/previewViewerHeaderComponentBase.tsx]: This change was in the other PR as well. Please try and minimize overlap between branches.

File: src/scripts/clipperUI/components/pdfPageSelectionRadioButton.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+import {Constants} from "../../constants";
+import {Localization} from "../../localization/localization";
+import {PdfPreviewInfo} from "../../previewInfo";
+import {Popover} from "./popover";
+import * as _ from "lodash";
+import {StringUtils} from "../../stringUtils";
+import {OperationResult} from "../../operationResult";
+import {ButtonGroupingsComponentBase, ControlGroup} from "./previewViewer/previewViewerHeaderComponentBase";
+
+class PdfPageSelectionRadioButton extends ButtonGroupingsComponentBase<{}, ClipperStateProp> {
```
Comment (mttwc) [inline src/scripts/clipperUI/components/pdfPageSelectionRadioButton.tsx]: Why would this extend the previously-named header class?

**Review by mttwc — CHANGES_REQUESTED** [review]

File: src/scripts/clipperUI/components/pdfPageSelectionRadioButton.tsx [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+			<div role="radio" id={Constants.Ids.radioAllPagesLabel} className="pdf-control" {...this.enableInvoke(this.onSelectionChange, pdfPreviewInfo.allPages ? selectedTabIndex : unselectedTabIndex, true, undefined, Constants.AriaSet.pdfPageSelection)}>
+				<div className={"pdf-indicator pdf-radio-indicator"}>
+					{pdfPreviewInfo.allPages ? <div className={Constants.Classes.radioIndicatorFill}></div> : undefined}
+				</div>
+				<div className="pdf-label-margin">
+					<span
+						className={"pdf-label" + (pdfPreviewInfo.allPages ? " focused" : "")}>{Localization.getLocalizedString("WebClipper.Label.PdfAllPagesRadioButton")}</span>
+				</div>
+			</div>
+			);
```
Comment (mttwc) [inline src/scripts/clipperUI/components/pdfPageSelectionRadioButton.tsx]: nit: Indentation

File: src/scripts/clipperUI/components/pdfPageSelectionRadioButton.tsx [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+		if (parsePageRangeOperation.status === OperationResult.Succeeded) {
+			throw Error("Given that shouldShowPopover is true, parsing the pageRange should never succeed: PageRange: " + pdfPreviewInfo.selectedPageRange);
+		}
+
+		return Localization.getLocalizedString("WebClipper.Popover.PdfInvalidPageRange").replace("{0}", parsePageRangeOperation.result as string);
+	}
+
+	getAllPagesButton(): any {
+		let pdfPreviewInfo = this.props.clipperState.pdfPreviewInfo;
+		let selectedTabIndex = 60;
```
Comment (mttwc) [inline src/scripts/clipperUI/components/pdfPageSelectionRadioButton.tsx]: The value `60` is duplicated below. Move this to the class level.

File: src/tests/clipperUI/components/pdfClipOptions_tests.tsx [hunk 6]
```diff
@@ -99,11 +99,19 @@ export class PdfClipOptionsTests extends TestModule {
 
 		test("The tab order should flow linearly between pdf options", () => {
```
Comment (mttwc) [inline src/tests/clipperUI/components/pdfClipOptions_tests.tsx]: Update test description

File: src/tests/clipperUI/mainController_tests.tsx [hunk 7]
```diff
@@ -77,11 +77,19 @@ export class MainControllerTests extends TestModule {
 			let controllerInstance = MithrilUtils.mountToFixture(this.defaultComponent);
```
Comment (mttwc) [inline src/tests/clipperUI/mainController_tests.tsx:77]: Update test description

File: src/tests/clipperUI/components/pdfClipOptions_tests.tsx [hunk 8]
```diff
@@ -99,11 +99,19 @@ export class PdfClipOptionsTests extends TestModule {
...
 			});
 			Assert.tabOrderIsIncremental([
-				Constants.Ids.radioAllPagesLabel, Constants.Ids.radioPageRangeLabel, Constants.Ids.checkboxToDistributePages, Constants.Ids.checkboxToAttachPdf
+				Constants.Ids.radioAllPagesLabel, Constants.Ids.checkboxToDistributePages, Constants.Ids.checkboxToAttachPdf
+			]);
+
+			MithrilUtils.simulateAction(() => {
+				document.getElementById(Constants.Ids.radioPageRangeLabel).click();
+			});
+			Assert.tabOrderIsIncremental([Constants.Ids.radioPageRangeLabel, Constants.Ids.checkboxToDistributePages, Constants.Ids.checkboxToAttachPdf
```
Comment (mttwc) [inline src/tests/clipperUI/components/pdfClipOptions_tests.tsx]: nit: style

File: src/tests/clipperUI/mainController_tests.tsx [hunk 9]
```diff
@@ -77,11 +77,19 @@ export class MainControllerTests extends TestModule {
...
 
 			MithrilUtils.simulateAction(() => {
-				controllerInstance.state.currentPanel = PanelType.ClipOptions;
```
Comment (mttwc) [inline src/tests/clipperUI/mainController_tests.tsx]: Why was this line removed?

**Review by mttwc — CHANGES_REQUESTED** [review]

File: src/tests/clipperUI/components/pdfClipOptions_tests.tsx (hunk 8 above)
Comment (JaneHazen) [reply src/tests/clipperUI/components/pdfClipOptions_tests.tsx]: ?

File: src/scripts/clipperUI/components/pdfPageSelectionRadioButton.tsx [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+		return (
+			<div role="radio" id={Constants.Ids.radioAllPagesLabel} className="pdf-control" {...this.enableInvoke(this.onSelectionChange, pdfPreviewInfo.allPages ? selectedTabIndex : unselectedTabIndex, true, undefined, Constants.AriaSet.pdfPageSelection)}>
+				<div className={"pdf-indicator pdf-radio-indicator"}>
+					{pdfPreviewInfo.allPages ? <div className={Constants.Classes.radioIndicatorFill}></div> : undefined}
+				</div>
+				<div className="pdf-label-margin">
+					<span className={"pdf-label" + (pdfPreviewInfo.allPages ? " focused" : "")}>{Localization.getLocalizedString("WebClipper.Label.PdfAllPagesRadioButton")}</span>
+				</div>
+			</div>
+			);
```
Comment (mttwc) [inline src/scripts/clipperUI/components/pdfPageSelectionRadioButton.tsx]: nit: indentation

**Review by mttwc — APPROVED** [review]

**Review by mttwc — APPROVED** [review]

**Pull request merged**